### PR TITLE
BME-40 Swapping ServerLevelMixin for ChunkHolderMixin and making ChunkHolderMixin work

### PR DESCRIPTION
**Pull request by maddie-j**
As per Discord:

> Looking back at ChunkHolder, it seems the function which calls broadcastChanges does so once per tick for every loaded chunk. This means we know two things about this function:
> 1. It will only be called once per tick.
> 2. It will only be called for loaded chunks.

> This should be significantly better than ServerLevel.onBlockStateChange which runs for every block change, being called multiple times for the same chunk.

> It also means we shouldn't have the same problem as the ChunkDataEvent.Save which resulted in unloaded chunks triggering updates. Speaking of, the way to tell if a chunk is loaded on the server side is apparently to check chunkholder.getTickingChunk() != null , in case anyone needs that in the future.

> Additionally, we should be able to check just this.hasChangedSections to see if any blocks changed in that chunk, cutting out updates from the previous version which only affected light updates. That should be an O(1) check, keeping things fast even if injected at the function HEAD.

Doing testing, this version of `ChunkHolderMixin` fires significantly fewer times per tick than `ServerLevelMixin` did, especially on events causing a lot of block changes such as tree growth or pistons in a redstone machine. Additionally, `ChunkHolderMixin` doesn't allocate any new objects, unlike `ServerLevelMixin`. This should make this setup more efficient than the previous build.

Separatley, I've included my debugging event logger but left its initialiser commented out. This has been very handy being able to cross reference logs from the thing I'm trying to measure with what happened in the world before and after that thing. Figured it's worth keeping the file around so it can be turned on and off as needed without having to rewrite it.

### CODE REVIEW

**Comment by LordFokas** [comment]
I'll be holding off on merging this until we properly profile the performance of both solutions.

**Comment by maddie-j** [comment]
As per the Discord, have tested both ChunkHolderMixin and ServerLevelMixin using the profiler. While the original ChunkHolderMixin proved less efficient than ServerLevelMixin, a modified version of ChunkHolderMixin that brough the mixin injector point within the existing changed chunks check cut down the most on unwanted triggering situations, leaving it (in my testing) more performant than ServerLevelMixin. 

I've also tidied up the test profiler to allow it to stay permanently and still make sense.

File: src/main/java/com/eerussianguy/blazemap/util/Profilers.java [hunk 1]
```diff
@@ -13,10 +13,8 @@ public static class Server {
         public static final TimeProfilerAsync TRANSFORMER_TIME_PROFILER = new TimeProfilerAsync(20);
         public static final LoadProfiler TRANSFORMER_LOAD_PROFILER = new LoadProfiler(20, 50);
 
-        public static final TimeProfilerSync CHUNKHOLDER_TIME_PROFILER = new TimeProfilerSync(20);
-        public static final LoadProfiler CHUNKHOLDER_LOAD_PROFILER = new LoadProfiler(20, 50);
-        public static final TimeProfilerSync SERVERLEVEL_TIME_PROFILER = new TimeProfilerSync(20);
-        public static final LoadProfiler SERVERLEVEL_LOAD_PROFILER = new LoadProfiler(20, 50);
+        public static final TimeProfilerSync TRIGGER_CHUNK_DIRTY_TIME_PROFILER = new TimeProfilerSync(20);
+        public static final LoadProfiler TRIGGER_CHUNK_DIRTY_LOAD_PROFILER = new LoadProfiler(20, 50);
```
Comment (LordFokas) [inline src/main/java/com/eerussianguy/blazemap/util/Profilers.java:17]: Please undo these (and correct its references) as I decided to keep the naming and I'll be reusing the ServerLevel ones for a different thing.

File: src/main/java/com/eerussianguy/blazemap/util/Profilers.java
Comment (LordFokas) [inline src/main/java/com/eerussianguy/blazemap/util/Profilers.java:1]: Please undo these (and correct its references) as I decided to keep the naming and I'll be reusing the ServerLevel ones for a different thing.

File: src/main/java/com/eerussianguy/blazemap/feature/ProfilingRenderer.java
Comment (LordFokas) [inline src/main/java/com/eerussianguy/blazemap/feature/ProfilingRenderer.java:1]: In line with the profilers, I'd like this file to stay unchanged as well. It'll have a dead profiler subsystem but I'm putting it to use soon.

**Comment by LordFokas** [comment]
Other than these 2 little things I'm ready to merge this.
Great job 😃

**Comment by maddie-j** [comment]
Done!

**Pull request merged**